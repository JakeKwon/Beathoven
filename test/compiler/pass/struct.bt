Struct sth {
    int a;
    string b;
    int c;
}

Struct sth s;
Struct sth t;
s.a = 17; // assign field
s.b = "struct string\n";
t = s; // copy structs
t.b = "struct s changed\n";
print(s.a);
print(s.b);
print(t.a);
print(t.b);
