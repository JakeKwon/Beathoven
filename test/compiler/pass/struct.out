17struct string
17struct s changed
